common/txmgr/confirmer.go

```diff
@@ -616,6 +625,24 @@ func (ec *Confirmer[CHAIN_ID, HEAD, ADDR, TX_HASH, BLOCK_HASH, R, SEQ, FEE]) sep
 	return
 }
 
+func (ec *Confirmer[CHAIN_ID, HEAD, ADDR, TX_HASH, BLOCK_HASH, R, SEQ, FEE]) resumeFailedTaskRuns(ctx context.Context, etx txmgrtypes.Tx[CHAIN_ID, ADDR, TX_HASH, BLOCK_HASH, SEQ, FEE]) error {
+	if !etx.PipelineTaskRunID.Valid || ec.resumeCallback == nil || !etx.SignalCallback {
+		return nil
+	}
+	err := ec.resumeCallback(ctx, etx.PipelineTaskRunID.UUID, nil, errors.New(ec.stuckTxDetector.StuckTxFatalError()))
+	if errors.Is(err, sql.ErrNoRows) {
+		ec.lggr.Debugw("callback missing or already resumed", "etxID", etx.ID)
```

Comment:
curious when would this happen ? also would it help by adding a check `etx.CallbackCompleted` on line 629 ?

Reply:
Good catch! Adding that check too. I think this could happen if we signal the callback but then the app goes down before we mark it as completed in our DB. When we start up again, we could assume it was never signaled and try the callback again.